kubernetes-client-openapi/src/main/java/io/micronaut/kubernetes/client/openapi/credential/ExecCommandCredentialLoader.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.kubernetes.client.openapi.credential;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.kubernetes.client.openapi.config.KubernetesClientConfiguration;
+import io.micronaut.kubernetes.client.openapi.config.model.ExecConfig;
+import io.micronaut.kubernetes.client.openapi.config.model.ExecEnvVar;
+import io.micronaut.kubernetes.client.openapi.config.KubeConfig;
+import io.micronaut.kubernetes.client.openapi.credential.model.ExecCredential;
+import io.micronaut.serde.ObjectMapper;
+import jakarta.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.InputStream;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The credential loader which uses the exec command from the kube config to get credentials.
+ */
+@Singleton
+@BootstrapContextCompatible
+class ExecCommandCredentialLoader implements KubernetesCredentialLoader {
```

Comment:
is there no other way to load credentials exception via using a process? Is this what the original K8s client does?

Reply:
There are multiple ways of authentication. One of those requires executing a command that is provided in the kube config file. For example:
```
users:
- name: user-c77rlwbu5ua
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      command: oci
      args:
      - ce
      - cluster
      - generate-token
      - --cluster-id
      - ocid1.cluster.oc1.phx.....
      - --region
      - us-phoenix-1
      env: []
```
or
```
users:
- name: arn:aws:eks:us-east-1:....
  user:
    exec:
      apiVersion: client.authentication.k8s.io/v1beta1
      command: aws
      args:
      - --region
      - us-east-1
      - eks
      - get-token
      - --cluster-name
      - cluster1
      - --output
      - json
```
And yes, the original K8s client uses a process and fabric one too